tests/extension/oneapi_enqueue_barrier/enqueue_barrier.cpp

```diff
@@ -0,0 +1,300 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+#include "../../common/common.h"
+
+namespace enqueue_barrier::tests {
+
+inline void skip_if_dev_does_not_support_usm_alloc(
+    sycl::device& dev, sycl::aspect usm_alloc_type) {
+  if (!dev.has(usm_alloc_type)) {
+    SKIP("Device doesn't support usm allocations");
+  }
+}
+
+constexpr size_t buffer_size = 10;
+constexpr size_t ker_a_iter_num = 1000000;
+constexpr size_t ker_b_iter_num = 3000000;
+
+class EnqueueBarrierTestBase {
+  template <size_t iter_num>
+  class enqueue_barrier_kernel {
+    using LoopArrAccT = sycl::accessor<int, 1, sycl::access_mode::write>;
+    using ResBufAccT = sycl::accessor<bool, 1, sycl::access_mode::write>;
+    LoopArrAccT loop_acc;
+    ResBufAccT res_acc;
+    bool* usm_ptr;
+
+   public:
+    enqueue_barrier_kernel(LoopArrAccT loop_arr_acc, ResBufAccT res_buf_acc,
+                           bool* flags)
+        : loop_acc(loop_arr_acc), res_acc(res_buf_acc), usm_ptr(flags){};
+    void operator()() const {
+      for (size_t i = 0; i < iter_num; i++) {
+        int val = sycl::sqrt(float(i));
+        loop_acc[val % buffer_size] = i;
+      }
+      res_acc[0] = usm_ptr[0] == false;
+      res_acc[1] = usm_ptr[1] == false;
+    }
+  };
+
+  virtual void submit_barrier(sycl::event& e1, sycl::event& e2,
+                              sycl::queue& q) = 0;
+
+ protected:
+  void run(sycl::queue& q1, sycl::queue& q2, sycl::queue& q3) {
+    bool* completion_flags = sycl::malloc_device<bool>(2, q1);
+
+    std::array result_in_ker_a{false, false};
+    std::array result_in_ker_b{false, false};
+
+    int loop_arr_a[buffer_size];
+    int loop_arr_b[buffer_size];
+
+    {
+      sycl::buffer<bool, 1> result_buf_a(result_in_ker_a.data(),
+                                         result_in_ker_a.size());
+      sycl::buffer<bool, 1> result_buf_b(result_in_ker_b.data(),
+                                         result_in_ker_b.size());
+      sycl::buffer<int, 1> loop_arr_buf_a(loop_arr_a, buffer_size);
+      sycl::buffer<int, 1> loop_arr_buf_b(loop_arr_b, buffer_size);
+
+      q1.submit([&](sycl::handler& cgh) {
+          cgh.single_task([=] {
+            completion_flags[0] = false;
+            completion_flags[1] = false;
+          });
+        }).wait();
+
+      auto event1 = q1.submit([&](sycl::handler& cgh) {
+        auto result_buf_a_acc =
+            sycl::accessor(result_buf_a, cgh, sycl::write_only);
+        auto loop_arr_buf_a_acc =
+            sycl::accessor(loop_arr_buf_a, cgh, sycl::write_only);
```

Comment:
While quite legal, why not simplifying everywhere with
```suggestion
        sycl::accessor result_buf_a_acc(result_buf_a, cgh, sycl::write_only);
        sycl::accessor loop_arr_buf_a_acc(loop_arr_buf_a, cgh, sycl::write_only);
```
?